janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/util/RestClientAuthenticatorBase.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.es.rest.util;
+
+import java.io.IOException;
+
+import org.apache.http.client.config.RequestConfig.Builder;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+
+/**
+ * Convenience class for using as parent when implementing custom REST client authenticator. By default
+ * this class does not do anything.
```

Comment:
"Base class for custom REST client authenticators."

Reply:
updated